From 12467e913923a62aa892ec9f3c49bf59e91c2c2a Mon Sep 17 00:00:00 2001
From: Satyeshwar Singh <satyeshwar.singh@intel.com>
Date: Sun, 14 Oct 2018 16:41:07 -0700
Subject: [PATCH 0987/1552] drm/i915: Check for a valid fb when computing
 watermarks

fb must be valid when we are computing watermarks for a plane or
else we will run into a crash. Checking to see if it is indeed
valid and returning EINVAL if it's not.

https://github.com/projectacrn/acrn-hypervisor/issues/1459
Signed-off-by: Satyeshwar Singh <satyeshwar.singh@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 36ba039..b218b1a 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4554,6 +4554,11 @@ skl_compute_plane_wm_params(const struct drm_i915_private *dev_priv,
 	if (!intel_wm_plane_visible(cstate, intel_pstate))
 		return 0;
 
+	if (plane_id == 1 && !fb) {
+		DRM_DEBUG_KMS("Invalid fb for plane\n");
+		return -EINVAL;
+	}
+
 	/* only NV12 format has two planes */
 	if (plane_id == 1 && fb->format->format != DRM_FORMAT_NV12) {
 		DRM_DEBUG_KMS("Non NV12 format have single plane\n");
-- 
2.7.4

